### PR TITLE
Spring 07

### PR DESCRIPTION
**Pull request by oroboro-cpu**

### CODE REVIEW

File: src/main/java/cinema/project/annotation/FieldsValueMatch.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    Class<?>[] groups() default {};
+
+    Class<? extends Payload>[] payload() default {};
+
+    @Target({ElementType.TYPE})
+    @Retention(RetentionPolicy.RUNTIME)
+    @interface List {
+        FieldsValueMatch[] value();
+    }
```
Comment (yurwing) [inline src/main/java/cinema/project/annotation/FieldsValueMatch.java]: you can delete it

File: src/main/java/cinema/project/annotation/LoginValidator.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class LoginValidator implements
+        ConstraintValidator<LoginConstraint, String> {
+
+    @Override
+    public void initialize(LoginConstraint contactNumber) {
+    }
```
Comment (yurwing) [inline src/main/java/cinema/project/annotation/LoginValidator.java]: if you don't use this method, don't add it

**Review by yurwing — CHANGES_REQUESTED** [review]

**Review by yurwing — APPROVED** [review]

File: src/main/java/cinema/project/annotation/FieldsValueMatch.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Constraint(validatedBy = FieldsValueMatchValidator.class)
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface FieldsValueMatch {
+
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/project/annotation/FieldsValueMatch.java]: redundant empty line

File: src/main/java/cinema/project/annotation/FieldsValueMatchValidator.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package cinema.project.annotation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldsValueMatchValidator
+        implements ConstraintValidator<FieldsValueMatch, Object> {
+
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/project/annotation/FieldsValueMatchValidator.java]: same

File: src/main/java/cinema/project/annotation/LoginValidator.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package cinema.project.annotation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class LoginValidator implements
+        ConstraintValidator<LoginConstraint, String> {
+    @Override
+    public boolean isValid(String contactField, ConstraintValidatorContext cxt) {
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/project/annotation/LoginValidator.java]: let's make parameter names more informative

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
let's implement p.2 in the HW
`Replace all ?userId @RequestParam in your code with Authentication object.`

also, remove salt from `User`

File: src/main/java/cinema/project/controller/OrderController.java [hunk 6]
```diff
@@ -32,15 +34,19 @@ public OrderController(OrderService service, UserService userService,
     }
 
     @GetMapping
-    public List<OrderResponseDto> getHistory(@RequestParam Long userId) {
-        return orderService.getOrdersHistory(userService.get(userId)).stream()
+    public List<OrderResponseDto> getHistory(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        User user = userService.findByEmail(userDetails.getUsername()).get();
+        return orderService.getOrdersHistory(userService.get(user.getId())).stream()
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/project/controller/OrderController.java]: can we simplify this line?

File: src/main/java/cinema/project/controller/OrderController.java [hunk 7]
```diff
@@ -32,15 +34,19 @@ public OrderController(OrderService service, UserService userService,
...
     }
 
     @PostMapping("/complete")
-    public void completeOrder(@RequestParam Long userId) {
+    public void completeOrder(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        User user = userService.findByEmail(userDetails.getUsername()).get();
         orderMapper.toDto(orderService.completeOrder(shoppingCartService
-                .getByUser(userService.get(userId))));
+                .getByUser(userService.get(user.getId()))));
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/project/controller/OrderController.java]: same

File: src/main/java/cinema/project/controller/ShoppingCartController.java [hunk 8]
```diff
@@ -32,13 +35,18 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
     }
 
     @GetMapping("/by-user")
-    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
-        return mapper.toDto(shoppingCartService.getByUser(userService.get(userId)));
+    public ShoppingCartResponseDto getByUser(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        User user = userService.findByEmail(userDetails.getUsername()).get();
+        return mapper.toDto(shoppingCartService.getByUser(userService.get(user.getId())));
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/project/controller/ShoppingCartController.java]: same

File: src/main/java/cinema/project/controller/ShoppingCartController.java [hunk 9]
```diff
@@ -32,13 +35,18 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
...
 
     @PostMapping("/movie-sessions")
-    public void addMovieSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+    public void addMovieSession(Authentication authentication,
+                                @RequestParam Long movieSessionId) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        User user = userService.findByEmail(userDetails.getUsername()).get();
         shoppingCartService.addSession(movieSessionService.get(movieSessionId),
-                userService.get(userId));
+                userService.get(user.getId()));
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/project/controller/ShoppingCartController.java]: same

File: src/main/java/cinema/project/annotation/LoginValidator.java [hunk 10]
```diff
@@ -6,7 +6,7 @@
 public class LoginValidator implements
         ConstraintValidator<LoginConstraint, String> {
     @Override
-    public boolean isValid(String contactField, ConstraintValidatorContext cxt) {
+    public boolean isValid(String contactField, ConstraintValidatorContext validatorContext) {
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/project/annotation/LoginValidator.java]: let's name in `login` instead of `contactField`

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**